### PR TITLE
reverse words in a string

### PR DESCRIPTION
**Pull request by karitarr**
Problem- Write a function that reverses the words in a string in place. 

*This problem requires a pretty big "a ha" moment that I probably wouldn't have gotten on my own. So I did get a hint after getting stuck using a much more complicated approach :)

@mLewisLogic @ianlogan @joshualee @lxdvs

### CODE REVIEW

File: reverse_words.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    right -= 1
+  end
+  return input_string
+
+end
+
+#loops through the entire string, keeping track of the start and end of the word and reversing them
+#one at a time
+def reverse_string!(input_string)
+  if input_string == ""
```
Comment (ianlogan) [inline reverse_words.rb]: Nice.

Comment (ianlogan) [inline reverse_words.rb]: Don't forget about nil ;)

Take a look at http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F

**Comment by karitarr** [comment]
is there a standard best practice (without using classes) for checking for object presence in a function?

**Comment by ianlogan** [comment]
It depends on the use case and it's up to the developer to decide what to do in each scenario - nil at times is expected and fine but others it is not.

**Comment by mLewisLogic** [comment]
:+1:

**Pull request merged**